### PR TITLE
feat(clients): generate code snippets from cts

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-1749

### Changes included:

This PR registers a new generator in order to generate code snippets for each method of each languages, it is completely based on the CTS request generator and will take the first test case of each method as the example.

The snippets are generated for
- java
- javascript
- python
- php

#### Generators change

I had to change the output folder and template dir in order to fully reuse the `TestsRequest` generator, which means each template has been edited to include the `tests` folder when using partial templates

#### clients.config.json

A new field named `snippets` allows us to store the output folder and the file extension, similarly to what we do for tests

#### CI

We now store the snippets in the artifacts and restore them in order to be pushed by the CI

### What's next

- Add snippets for missing clients
- Improve first test case variable values and descriptions

### CODE REVIEW

**Review by Fluf22 — APPROVED** [review]
Ooh nice

File: generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java [hunk 1]
```diff
@@ -63,6 +63,19 @@ public void processOpts() {
...
+    SnippetsGenerator snippetGenerator = new SnippetsGenerator(language, client);
+
+    if (!snippetGenerator.available()) {
+      return;
+    }
+
+    String snippetsExtension = Helpers.getClientConfigField(language, "snippets", "extension");
+    String snippetsOutputFolder = Helpers.getClientConfigField(language, "snippets", "outputFolder");
+
+    snippetGenerator.addSupportingFiles(supportingFiles, snippetsOutputFolder, snippetsExtension);
```
Comment (millotp) [inline generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java:76]: the `SnippetsGenerator` should do that logic itself

File: scripts/cts/generate.ts [hunk 2]
```diff
@@ -43,6 +43,9 @@ export async function ctsGenerateMany(generators: Generator[]): Promise<void> {
       await run('YARN_ENABLE_IMMUTABLE_INSTALLS=false yarn install', {
         cwd: 'tests/output/javascript',
       });
+      await run('YARN_ENABLE_IMMUTABLE_INSTALLS=false yarn install', {
```
Comment (millotp) [inline scripts/cts/generate.ts:46]: I think this should be in a totally different file, like `scripts/snippets/generate.ts`

**Review by millotp — COMMENTED** [review]
so cool, almost nothing to add !

File: generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java (hunk 1 above)
Comment (shortcuts) [reply generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java:76]: ah yes it can

File: scripts/cts/generate.ts (hunk 2 above)
Comment (shortcuts) [reply scripts/cts/generate.ts:46]: I did not yet created the CLI stuff, I know you want to do it :D

**Review by millotp — APPROVED** [review]
let's start with that !

**Pull request merged**